pandas/core/arrays/base.py

```diff
@@ -1452,6 +1448,31 @@ def _reduce(self, name: str, *, skipna: bool = True, **kwargs):
     # Non-Optimized Default Methods; in the case of the private methods here,
     #  these are not guaranteed to be stable across pandas versions.
 
+    def _hash_pandas_object(
+        self, *, encoding: str, hash_key: str, categorize: bool
+    ) -> npt.NDArray[np.uint64]:
+        """
+        Hook for hash_pandas_object.
+
+        Default is likely non-performant.
+
+        Parameters
+        ----------
+        encoding : str
+        hash_key : str
+        categorize : bool
+
+        Returns
+        -------
+        np.ndarray[uint64]
+        """
+        from pandas.core.util.hashing import hash_array
+
+        values = self.to_numpy(copy=False)
```

Comment:
@jbrockmendel is there a reason you didn't use `self._values_for_factorize()` as the values to hash (as default), which would preserve the current behaviour? 

In addition, if we want to allow EAs to override this, we should either document what the keyword arguments mean, and/or we should expose `hash_array` publicly (eg in `pandas.api.extensions`) ?

Reply:
>  is there a reason you didn't use self._values_for_factorize()

IIRC it was some combination of _values_for_factorize not being required and trying to respect the _vfc shouldn't be used for anything _but_ factorize policy.

> In addition, if we want to allow EAs to override this, we should either document what the keyword arguments mean, and/or we should expose hash_array publicly (eg in pandas.api.extensions) ?

Documenting makes sense.

Reply:
> IIRC it was some combination of _values_for_factorize not being required and trying to respect the _vfc shouldn't be used for anything but factorize policy.

I have said that before, but I don't see any problem with using that for both, as we have always done. They are very related, and typically values suitable for factorization will also be suitable for hashing, given that factorization is hash-based. And now with this new method, EAs can still override it if they want  to use something else than `_values_for_factorize`. 

We actually have a bunch of internal doc comments indicating that `_values_for_factorize` is also used for hash_pandas_object. And we can add this to the actual `_values_for_factorize` docstring in the base class to make this use case explicit.

For external EAs that relied on this, this is a regression that we no longer use this by default (e.g. for geopandas this will now fail in `hash_object_array`, falling back to casting to string before trying it again. But the string representation is not that faithful, and thus can give wrong hashes)

Reply:
> And we can add this to the actual `_values_for_factorize` docstring in the base class to make this use case explicit.

It _was_ actually documented, which was removed here.